Add dashboard for zookeeper (base-quorum)

francoisferrand opened the pull request:
Issue: ZENKO-3736

francoisferrand committed:
Add dashboard for zookeeper (base-quorum)

Issue: ZENKO-3736

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

lavigne958 commented on monitoring/dashboards/zookeeper.json:
```diff
@@ -0,0 +1,9030 @@
+        "multi": true,
+        "name": "pod",
+        "options": [],
+        "query": "label_values({job=~\"${job}\",namespace=~\"${namespace}\"}, pod)",
```
se should not need to use regexp to filter namespace or job. we already know the exact name and don't use regexp for matching, we could simply use `job=\"${job}\"`

francoisferrand replied:
not for instances (e..g `pod`) though, otherwise viewing "All" instances does not work.

lavigne958 commented on monitoring/dashboards/zookeeper.json:
```diff
@@ -0,0 +1,9030 @@
+  },
+  "timezone": "",
+  "title": "ZooKeeper",
+  "uid": "SDE76m7Zzz",
```
I think you tooke the dashboard from grafana, I mean: you import a dashboard then you check it in grafana then you copy/paste the section "JSON model", but this contains some _runtime_ attributes, like the UID, which we don't want to set, I think it is generated by grafana when you import your dashboard.

francoisferrand replied:
indeed, uid should not be set, good catch! Should we just remove it? Or set it to `null` ?

I am wondering also what we should do about `version`, `revision` and `iteration`...

lavigne958 replied:
`version`: 1, we increment as we update it
`revision`: can be removed I suppose, we have not interest in this one
`iteration`: same no interest I think, can be removed if grafana does not require it.

francoisferrand committed:
Avoid using regex for __inputs parameters

francoisferrand committed:
Use pod name to identify instance in dashboards

This more readable than using IP, and consistent with the variable
used for filtering.

francoisferrand committed:
Fix version & remove unused fields

uid and iteration fields are generated automatically, so set them to
null. Set version to the nmber of revisions of the file, as returned by:

```
git rev-list HEAD -- monitoring/dashboards/zookeeper.json | wc -l
```

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3736.

Goodbye francoisferrand.
